resources/js/oauth2/consent.js

```diff
@@ -0,0 +1,163 @@
+import React, {useRef} from "react";
+import ReactDOM from "react-dom";
+import Button from '@material-ui/core/Button';
+import Container from "@material-ui/core/Container";
+import CssBaseline from "@material-ui/core/CssBaseline";
+import {MuiThemeProvider, createTheme, withStyles} from "@material-ui/core/styles";
+import Card from "@material-ui/core/Card";
+import CardHeader from "@material-ui/core/CardHeader";
+import CardContent from "@material-ui/core/CardContent";
+import Divider from "@material-ui/core/Divider";
+import Grid from "@material-ui/core/Grid";
+import InfoOutlinedIcon from "@material-ui/icons/InfoOutlined";
+import Typography from "@material-ui/core/Typography";
+import Tooltip from '@material-ui/core/Tooltip';
+
+import styles from "./consent.module.scss";
+
+const HtmlTooltip = withStyles((theme) => ({
+    tooltip: {
+        backgroundColor: '#f5f5f9',
+        color: 'rgba(0, 0, 0, 0.87)',
+        maxWidth: 220,
+        fontSize: theme.typography.pxToRem(12),
+        border: '1px solid #dadde9',
+    },
+}))(Tooltip);
+
+const ConsentPage = ({appLogo, appName, appDescription, csrfToken, disclaimer, requestedScopes, formAction}) => {
+    const formEl = useRef(null);
+    const trustEl = useRef(null);
+
+    const handleAccept = (e) => {
+        trustEl.current.value = 'AllowOnce';
+        formEl.current.submit();
+        e.preventDefault();
+    }
+
+    const handleCancel = (e) => {
+        trustEl.current.value = 'DenyOnce';
+        formEl.current.submit();
+        e.preventDefault();
+    }
+
+    return (
+        <Container component="main" maxWidth="xs" className={styles.main_container}>
+            <CssBaseline/>
+            <div className={styles.title_container}>
+                <a href={window.location.href} target='_self'>
+                    <img className={styles.app_logo} alt="appLogo" src={appLogo}/>
+                </a>
+                <h1>{appName}&nbsp;
+                    <HtmlTooltip
+                        arrow
+                        title={
+                            <>
+                                <Typography color="inherit">{appName}</Typography>
+                                {appDescription}
```

Comment:
app description should be the oauth2 app description if it set

Reply:
also please re add the missing text

<a target='_blank' href="{!!$website!!}">{!!$app_name!!}</a>&nbsp;<span style="font-size: 10pt;" class="glyphicon glyphicon-info-sign pointable info" aria-hidden="true" data-content="Contact Email: <a href='mailto:{!!$contact_email!!}'>{!!$contact_email!!}</a>.<br> Clicking 'Accept' will redirect you to: <a href='{!!$website!!}' target='_blank'>{!!$website!!}</a>
that is all info related to the oauth2 app itsself

Reply:
Done